# Z score testing

In [44]:
import pandas as pd
import numpy as np
import os
import sys
import csv

## Hypothesis Statement
#### Null Hypothesis:
The average commute time of the new bus is not significantly shorter than or equal to the existing X8 bus. 

# _$H_0$_ : $T_{\mathrm{new}} <= T_{\mathrm{old}} $
# _$H_1$_ : $T_{\mathrm{new}} > T_{\mathrm{old}} $
I set the significance level as: $\alpha=0.05$


In [45]:
url = "https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt"

In [46]:
# Read txt as csv:
    #Reference link: https://stackoverflow.com/questions/21546739/load-data-from-txt-with-pandas
    #Pandas.read_csv properties: https://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html
Bus = pd.read_csv(url, sep=",", names=['Commute Time'])

In [47]:
Bus.head()

Commute Time
0     31.622239
1     32.821376
2     30.229101
3     31.413766
4     39.010550

In [48]:
m = Bus['Commute Time'].mean()
std = Bus['Commute Time'].std()
N = len(Bus)
print ("Mean:", m, "\nStandard Deviation:", std, "\nN", N)

Mean: 34.4661616883 
Standard Deviation: 7.13728013375 
N 100


In [49]:
# Population Mean and Standard Deviation statement
MeanP = 36
StDev = 6

In [50]:
# Calculate Z-score
Num = MeanP-m
Den = StDev/np.sqrt(N)
Z = Num/Den
print('Z-test result:', Z)

Z-test result: 2.55639718617


### Result interpretation:
Z test results shows that 2.556 is larger than 2; as I originally chose 0.05 as the significance level which corresponds to 2 sigma. Therefore, the null hypothesis is rejected. In further, it means this new bus route is possibly relatively efficient and shoule possibly be implemented soon.